CMakeLists.txt

```diff
@@ -135,12 +133,12 @@ list(APPEND EXTRA_LIBRARIES ${CMAKE_DL_LIBS})
 # this conflicts with mpi4py-1.3 (is used to check for MPI version there)
 # add_definitions( -DHAVE_CONFIG_H )
 
+# Minimum required Boost version is set to 1.58 (only because of travis)
+# but the version in contrib is 1.61;
+
 if (EXTERNAL_BOOST)
-  set(NEEDED Boost_VERSION 1.55.0)
-  if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5")
-    set(NEEDED_Boost_VERSION 1.58.0) #issue #8
-  endif()
-  find_package(Boost ${NEEDED_Boost_VERSION} COMPONENTS mpi python serialization system filesystem)
+  set(NEEDED_Boost_VERSION 1.58.0)
+  find_package(Boost ${NEEDED_Boost_VERSION} REQUIRED COMPONENTS mpi python serialization system filesystem)
```

Comment:
This variable is also used in error message (line 143) so I think we can leave it; 
the other thing is that travis does not have Boost 1.58 ... I still have to look into it and I hope to not build it everytime;